### PR TITLE
Adds redesign for the customize gas advanced tab.

### PR DESCRIPTION
**Pull request by danjm**
~- Let's review and merge https://github.com/MetaMask/metamask-extension/pull/5254 first~
~- Tests in progress.~
- The gas price chart component is just a placeholder for now. Implementation in a future PR.

Screenshots:

screenshot from 2018-09-21 00-09-05
----
screenshot from 2018-09-21 00-09-23

### CODE REVIEW

File: ui/app/components/gas-customization/gas-modal-page-container/advanced-tab-content/tests/advanced-tab-content-component.test.js [hunk 1]
```diff
@@ -40,12 +46,31 @@ describe('AdvancedTabContent Component', function () {
...
+      assert.deepEqual(renderDataSummaryArgs, ['$0.25', 21500])
+
+      assert.equal(AdvancedTabContent.prototype.renderGasEditRows.callCount, 1)
+      const renderGasEditRowArgs = AdvancedTabContent.prototype.renderGasEditRows.getCall(0).args
+      assert.deepEqual(renderGasEditRowArgs, [
+        11, propsMethodSpies.updateCustomGasPrice, 23456, propsMethodSpies.updateCustomGasLimit,
+      ])
+    })
+
+    it('should call renderGasEditRows with the expected params', () => {
```
Comment (whymarrh) [inline ui/app/components/gas-customization/gas-modal-page-container/advanced-tab-content/tests/advanced-tab-content-component.test.js]: This test is empty, can we remove it?

File: ui/app/components/gas-customization/gas-modal-page-container/gas-modal-page-container.component.js [hunk 2]
```diff
@@ -51,32 +52,34 @@ export default class GasModalPageContainer extends Component {
         updateCustomGasLimit={convertThenUpdateCustomGasLimit}
         customGasPrice={customGasPrice}
         customGasLimit={customGasLimit}
-        millisecondsRemaining={91000}
+        timeRemaining={'1 min 31 sec'}
```
Comment (whymarrh) [inline ui/app/components/gas-customization/gas-modal-page-container/gas-modal-page-container.component.js]: This should be `timeRemaining="1 min 31 sec"` (without the braces)

File: ui/app/components/gas-customization/gas-modal-page-container/gas-modal-page-container.component.js [hunk 3]
```diff
@@ -51,32 +52,34 @@ export default class GasModalPageContainer extends Component {
...
+  renderInfoRows (newTotalFiat, newTotalEth, sendAmount, transactionFee) {
+    const baseClassName = 'gas-modal-content__info-row'
 
-  renderInfoRows (originalTotalFiat, originalTotalEth, newTotalFiat, newTotalEth) {
     return (
       <div>
-        { this.renderInfoRow('gas-modal-content__info-row--fade', 'originalTotal', originalTotalFiat, originalTotalEth) }
-        { this.renderInfoRow('gas-modal-content__info-row', 'newTotal', newTotalFiat, newTotalEth) }
+        <div className={baseClassName}>
+          <div className={`${baseClassName}__send-info`}>
```
Comment (whymarrh) [inline ui/app/components/gas-customization/gas-modal-page-container/gas-modal-page-container.component.js]: Why are we doing string interpolation here on these class names? Is this something we've done in other places?

**Review by whymarrh — DISMISSED** [review]

**Comment by danjm** [comment]
Styling issues have been addressed in this downstream PR: https://github.com/MetaMask/metamask-extension/pull/5378

File: ui/app/components/page-container/page-container-footer/page-container-footer.component.js [hunk 4]
```diff
@@ -25,19 +26,20 @@ export default class PageContainerFooter extends Component {
...
       disabled,
       submitButtonType,
+      hideCancel,
     } = this.props
 
     return (
       <div className="page-container__footer">
 
-        <Button
+        {!hideCancel && <Button
```
Comment (alextsg) [inline ui/app/components/page-container/page-container-footer/page-container-footer.component.js]: Would it be enough to check for an `onCancel` function prop here? Or would you prefer to explicitly have a `hideCancel` prop?

Comment (danjm) [reply ui/app/components/page-container/page-container-footer/page-container-footer.component.js]: I mostly just did this as extra assurance that my change would be backwards compatible. I guess I am partial to explicitly using a `hideCancel` prop; the separation between code responsible for presentation and code responsible for handling user action feels cleaner that way.

File: ui/app/components/gas-customization/gas-modal-page-container/advanced-tab-content/advanced-tab-content.component.js [hunk 5]
```diff
@@ -82,25 +78,24 @@ export default class AdvancedTabContent extends Component {
...
       customGasLimit,
       totalFee,
     } = this.props
 
     return (
       <div className="advanced-tab">
-        { this.renderDataSummary(totalFee, millisecondsRemaining) }
-        <div className="advanced-tab__fee-chart-title">
-          { this.context.t('feeChartTitle') }
+        { this.renderDataSummary(totalFee, timeRemaining) }
```
Comment (alextsg) [inline ui/app/components/gas-customization/gas-modal-page-container/advanced-tab-content/advanced-tab-content.component.js]: I notice that you like accessing `this.props` in `render` and passing them to other methods in the same class. Do you notice benefits in doing that? Just curious.

**Review by alextsg — APPROVED** [review]

**Pull request merged**